src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1512,17 +1512,27 @@ def get_eventual_alias(self) -> AliasInfo:
                 segment containing it, and whether it's an alias.
 
         """
-        alias_expression = self.get_child("alias_expression")
+        # Get any table expressions
         tbl_expression = self.get_child("table_expression")
         if not tbl_expression:  # pragma: no cover
-            tbl_expression = self.get_child("bracketed").get_child("table_expression")
+            _bracketed = self.get_child("bracketed")
+            if _bracketed:
+                tbl_expression = _bracketed.get_child("table_expression")
         # For TSQL nested, bracketed tables get the first table as reference
         if tbl_expression and not tbl_expression.get_child("object_reference"):
-            if tbl_expression.get_child("bracketed"):
-                tbl_expression = tbl_expression.get_child("bracketed").get_child(
-                    "table_expression"
-                )
-        ref = tbl_expression.get_child("object_reference") if tbl_expression else None
+            _bracketed = tbl_expression.get_child("bracketed")
+            if _bracketed:
```

Comment:
Based on the previous behaviour, i wonder if this should be an assert?

Reply:
Based on my reading of the old behaviour it only tries to `.get_child("table_expression")` if it's already confirmed that `tbl_expression.get_child("bracketed")` is non null (albeit in an inefficient way by calling `tbl_expression.get_child("bracketed")` twice).

That means it did previously allow `tbl_expression.get_child("bracketed")` to be `None`. Did you mean something else @WittierDinosaur ?

Reply:
Ah, I just completely read the old code wrong - you're right